src/ActionKit/ActionRunner.php

```diff
@@ -107,7 +111,50 @@ public function run($actionName, $arguments = array() )
             return $this->results[ $actionName ] = $action->getResult();
         }
 
-        throw new Exception( "Can not create action class $class" );
+        throw new UnableToCreateActionException( "Can not create action class $class" );
+    }
+
+    public function runWith($request) 
+    {
+        if (!$request->getActionName()) {
+            throw new InvalidActionNameException("");
+        }
+        if ( ! Utils::validateActionName( $request->getActionName() ) ) {
```

Comment:
throw exception from ActionRequest constructor? 
if throw exception from ActionRequest constructor, $request will be undefined variable in L148 (if ( $request->isAjax() ) {...} ).